frontend/fingertips-frontend/components/views/OneIndicatorOneAreaView/index.tsx

```diff
@@ -0,0 +1,33 @@
+import { OneIndicatorOneAreaDashboard } from '@/components/dashboards/OneIndicatorOneAreaDashboard';
+import { areaCodeForEngland } from '@/lib/chartHelpers/constants';
+import {
+  SearchParams,
+  SearchStateManager,
+  SearchStateParams,
+} from '@/lib/searchStateManager';
+
+type OneIndicatorOneAreaViewProps = {
+  searchState: SearchStateParams;
+};
+
+export default async function OneIndicatorOneAreaView({
+  searchState,
+}: Readonly<OneIndicatorOneAreaViewProps>) {
+  const stateManager = SearchStateManager.initialise(searchState);
+  const {
+    [SearchParams.AreasSelected]: areaCodes,
+    [SearchParams.GroupSelected]: selectedGroupCode,
+  } = stateManager.getSearchState();
+
+  const areasSelected = areaCodes ?? [];
+
+  const areaCodesToRequest =
+    selectedGroupCode && selectedGroupCode != areaCodeForEngland
+      ? [...areasSelected, areaCodeForEngland, selectedGroupCode]
+      : [...areasSelected, areaCodeForEngland];
```

Comment:
We know this should only be a single value, so we could do something like:
```suggestion
  if (areaCodes?.length !== 1) {
    throw new Error('Invalid parameters provided to view');
  }

  const areaSelected = areaCodes[0];

  const areaCodesToRequest = [areaSelected, areaCodeForEngland];
  if (selectedGroupCode && selectedGroupCode != areaCodeForEngland) {
    areaCodesToRequest.push(selectedGroupCode);
  }
```

Given it's fundamentally incorrect to create this component with the wrong number of area codes we should verify that up front and throw an error if we've not got data we should have. Then (because typescript is really nice about narrowing types) we can continue from that point safely knowing that the parameters are as we expect. We should also extend this to throwing an error when anything other than a single indicator is passed, when we need to read those.

Reply:
Thanks for the suggestion, that will pattern across several views.